# Projeto Candidato x Investimento

***

Este notebook tem o intuito de <b>estudar uma possível correlação entre financiamento eleitoral e numero de votos</b> obtidos por candidatos em eleições diretas. O estudo será feito nas seguintes etapas:


<b><font color=green> 1. Importando os dados.</b></font>

<b><font color=green> 2. Analise das variáveis.</b></font>


Abaixo seguem as etapas detalhadas:



***

## 1. <font color=red>IMPORTANDO OS DADOS</font>

In [1]:
# importando as possíveis bibliotecas a serem utilizadas

import matplotlib.pyplot as plt
%matplotlib inline

import pandas as pd
import numpy as np
import seaborn as sns

### 1.1 <font color=green>Importando os dados das receitas dos candidatos</font>

***

In [2]:
# localizando o arquivo no disco rígido

local_arquivo_receitas = "dados/receitas_candidato-07112022.csv" 

In [3]:
# criando um dataframe com os dados baixados

receitas_candidatos_SP = pd.read_csv(local_arquivo_receitas, sep=',')

In [4]:
# visualizando os dados baixados

receitas_candidatos_SP.head()

ano  turno       tipo_eleicao sigla_uf  id_municipio  id_municipio_tse  \
0  2020      1  eleicao ordinaria       SP       3550308             71072   
1  2020      1  eleicao ordinaria       SP       3538709             68756   
2  2020      1  eleicao ordinaria       SP       3518800             64777   
3  2020      1  eleicao ordinaria       SP       3529401             66893   
4  2020      1  eleicao ordinaria       SP       3510609             63134   

   numero_candidato  cpf_candidato  cnpj_candidato  titulo_eleitor_candidato  \
0             45678     7902417815             NaN                       NaN   
1             25010     4984729813             NaN                       NaN   
2             13217    87471779853             NaN                       NaN   
3             12600    30550214879             NaN                       NaN   
4             19619    17750686843             NaN                       NaN   

   ...  cpf_administrador  numero_recibo_eleitoral numero_documento  \
0  ...                NaN                      NaN              NaN   
1  ...                NaN                      NaN              NaN   
2  ...                NaN                      NaN              NaN   
3  ...                NaN                      NaN              NaN   
4  ...                NaN                      NaN              NaN   

    numero_recibo_doacao  numero_documento_doacao tipo_prestacao_contas  \
0  456781371072SP000002E                      NaN                 final   
1  250101368756SP000001E                591366482                 final   
2  132171364777SP000001E                      NaN                 final   
3  126001366893SP000007E                      NaN                 final   
4  196191363134SP000003E                      NaN                 final   

  data_prestacao_contas sequencial_prestador_contas  cnpj_prestador_contas  \
0            2020-12-09                  1845020286         38838436000127   
1            2020-12-15                  1828971303         38510704000187   
2            2020-12-04                  1832935796         38569259000120   
3            2020-12-10                  1844928372         38843040000178   
4            2020-12-02                  1844567089         38809101000180   

  entrega_conjunto  
0              NaN  
1              NaN  
2              NaN  
3              NaN  
4              NaN  

[5 rows x 57 columns]

In [5]:
# verificando as variáveis dos dados coletados

receitas_candidatos_SP.columns

Index(['ano', 'turno', 'tipo_eleicao', 'sigla_uf', 'id_municipio',
       'id_municipio_tse', 'numero_candidato', 'cpf_candidato',
       'cnpj_candidato', 'titulo_eleitor_candidato', 'sequencial_candidato',
       'id_candidato_bd', 'nome_candidato', 'cpf_vice_suplente',
       'numero_partido', 'nome_partido', 'sigla_partido', 'cargo',
       'sequencial_receita', 'data_receita', 'fonte_receita', 'origem_receita',
       'natureza_receita', 'especie_receita', 'situacao_receita',
       'descricao_receita', 'valor_receita', 'sequencial_candidato_doador',
       'cpf_cnpj_doador', 'sigla_uf_doador', 'id_municipio_tse_doador',
       'nome_doador', 'nome_doador_rf', 'cargo_candidato_doador',
       'numero_partido_doador', 'sigla_partido_doador', 'nome_partido_doador',
       'esfera_partidaria_doador', 'numero_candidato_doador', 'cnae_2_doador',
       'descricao_cnae_2_doador', 'cpf_cnpj_doador_orig', 'nome_doador_orig',
       'nome_doador_orig_rf', 'tipo_doador_orig',
       'descri

In [6]:
receitas_candidatos_SP.turno.unique()

array([1, 2], dtype=int64)

In [7]:
# selecionando as variáveis para estudo

selecao_receitas_candidatos_SP = receitas_candidatos_SP[
    ['id_municipio_tse', # identificação do município
     'turno', # identificando o turno da eleição
     'numero_candidato', 'nome_candidato', 'cargo', # identificação do candidato
     'nome_partido', 'sigla_partido', # identificação do partido do candidato
     'descricao_receita', 'valor_receita']] # identificação da receita

selecao_receitas_candidatos_SP.head()

id_municipio_tse  turno  numero_candidato  \
0             71072      1             45678   
1             68756      1             25010   
2             64777      1             13217   
3             66893      1             12600   
4             63134      1             19619   

                      nome_candidato     cargo  \
0             GILSON ALMEIDA BARRETO  vereador   
1                   DILNEI DE CAMPOS  vereador   
2  GIZELIA APARECIDA DA SILVA ARAUJO  vereador   
3        ALESSANDRO FAUSTINO MARTINS  vereador   
4                 CESAR AUGUSTO JOSE  vereador   

                              nome_partido sigla_partido  \
0  Partido da Social Democracia Brasileira          PSDB   
1                               Democratas           DEM   
2                Partido dos Trabalhadores            PT   
3          Partido Democrático Trabalhista           PDT   
4                                  Podemos          PODE   

                                   descricao_receita  valor_receita  
0        IMÕVEL NA AVENIDA MATEO BEI 2614 SAO MATEUS        20000.0  
1                                                NaN          200.0  
2         VEICULO MONZA CINZA ANO 1983 PLACA BLZ4104         1000.0  
3                  FIAT PALIO FIRE ANO/MOD 2012/2013         1000.0  
4  CESSÃO DE VEÍCULO PRÓPRIO PARA USO DURANTE A C...         2000.0

In [8]:
# verificando o tamanho do dataframe

selecao_receitas_candidatos_SP.shape

(354934, 9)

In [9]:
# verificando os cargos para o estado de SP com os dados colhidos

selecao_receitas_candidatos_SP.cargo.unique()

array(['vereador', 'prefeito', 'vice-prefeito'], dtype=object)

In [10]:
# escolhendo somente os dados para as prefeituras do estado de SP

selecao_prefeitos_receitas_candidatos_SP = selecao_receitas_candidatos_SP.query('cargo == "prefeito"')

# vizualizando os dados coletados
selecao_prefeitos_receitas_candidatos_SP.head()

id_municipio_tse  turno  numero_candidato  \
5              63134      1                45   
6              68578      1                13   
13             62910      1                23   
26             68098      1                45   
27             63894      1                13   

                      nome_candidato     cargo  \
5     MARCO AURELIO DOS SANTOS NEVES  prefeito   
6             GEREMIAS RIBEIRO PINTO  prefeito   
13  VANDRÉ LUIZ DE CAMARGO VON ZUBEN  prefeito   
26                CARLOS HIROCI OUTI  prefeito   
27             CÉLIA REGINA BRANDANI  prefeito   

                               nome_partido sigla_partido  \
5   Partido da Social Democracia Brasileira          PSDB   
6                 Partido dos Trabalhadores            PT   
13                                Cidadania     CIDADANIA   
26  Partido da Social Democracia Brasileira          PSDB   
27                Partido dos Trabalhadores            PT   

                                descricao_receita  valor_receita  
5                         VENTILADORES VENTISILVA         178.03  
6   FORD FOCUS SEDAN 2010/2011 EM NOME DO CONJUGE        1000.00  
13                                            NaN       20000.00  
26                                            NaN       12000.00  
27                              MESA DE PVC USADA          55.00

### 1.2 <font color=green>Importando os dados dos códigos dos municípios</font>

***

In [11]:
# localizando o arquivo no disco rígido

local_arquivo_municipios_tse = "dados/municipios_brasileiros_tse-07112022.csv" 

In [14]:
# criando um dataframe com os dados baixados
municipios_codigo_tse = pd.read_csv(local_arquivo_municipios_tse, sep=',')

# visualizando os dados coletados
municipios_codigo_tse.head()

codigo_tse  uf nome_municipio  capital  codigo_ibge
0        1120  AC     ACRELÂNDIA        0      1200013
1        1570  AC   ASSIS BRASIL        0      1200054
2        1058  AC      BRASILÉIA        0      1200104
3        1007  AC         BUJARI        0      1200138
4        1015  AC       CAPIXABA        0      1200179

In [15]:
# localizando o arquivo no disco rígido

municipios_SP_codigo_tse = municipios_codigo_tse.query('uf == "SP"').reset_index(drop=True)

# visualizando o dataframe com o código do tse para os municípios do estado de SP
municipios_SP_codigo_tse

codigo_tse  uf      nome_municipio  capital  codigo_ibge
0         61018  SP          ADAMANTINA        0      3500105
1         61034  SP              ADOLFO        0      3500204
2         61050  SP               AGUAÍ        0      3500303
3         61131  SP              AGUDOS        0      3500709
4         61581  SP            ALAMBARI        0      3500758
..          ...  ..                 ...      ...          ...
640       61115  SP  ÁGUAS DE SÃO PEDRO        0      3500600
641       61239  SP    ÁLVARES FLORENCE        0      3501202
642       61255  SP     ÁLVARES MACHADO        0      3501301
643       61271  SP  ÁLVARO DE CARVALHO        0      3501400
644       67776  SP                ÓLEO        0      3533809

[645 rows x 5 columns]

### 1.3 <font color=green>Importando os dados dos resultados das votações de cada candidato</font>

***

In [16]:
# localizando o arquivo no disco rígido

local_arquivo_resultados = "dados/resultados_candidatos-07112022.csv" 

In [17]:
# criando um dataframe com os dados baixados

resultados_candidatos_SP = pd.read_csv(local_arquivo_resultados, sep=',')

# visualizando os dados coletados dos resultados de cada candidato no estado de SP
resultados_candidatos_SP.head()

ano  turno       tipo_eleicao sigla_uf  id_municipio  id_municipio_tse  \
0  2020      1  eleicao ordinaria       SP       3500105             61018   
1  2020      1  eleicao ordinaria       SP       3500105             61018   
2  2020      1  eleicao ordinaria       SP       3500105             61018   
3  2020      1  eleicao ordinaria       SP       3500105             61018   
4  2020      1  eleicao ordinaria       SP       3500105             61018   

      cargo  numero_partido sigla_partido  numero_candidato  \
0  prefeito              25           DEM                25   
1  prefeito              55           PSD                55   
2  vereador              10  REPUBLICANOS             10999   
3  vereador              10  REPUBLICANOS             10242   
4  vereador              10  REPUBLICANOS             10333   

   sequencial_candidato  id_candidato_bd   resultado  votos  
0          250001018914        1018939.0      eleito  11863  
1          250001085198        1319672.0  nao eleito   3367  
2          250001023020         968597.0  nao eleito     28  
3          250001023021        1422654.0  nao eleito     33  
4          250001023022        1730446.0  nao eleito    353

In [18]:
# verificando as variáveis dos dados coletados

resultados_candidatos_SP.columns

Index(['ano', 'turno', 'tipo_eleicao', 'sigla_uf', 'id_municipio',
       'id_municipio_tse', 'cargo', 'numero_partido', 'sigla_partido',
       'numero_candidato', 'sequencial_candidato', 'id_candidato_bd',
       'resultado', 'votos'],
      dtype='object')

In [19]:
# selecionando as variáveis para estudo

selecao_resultados_candidatos_SP = resultados_candidatos_SP[
    ['id_municipio_tse', # identificação do município
     'numero_candidato', # identificação do candidato
     'turno', # identificando o turno da eleição
     'sigla_partido', 'cargo', # identificação do partido do candidato
     'resultado', 'votos']] # identificação da receita

selecao_resultados_candidatos_SP.head()

id_municipio_tse  numero_candidato  turno sigla_partido     cargo  \
0             61018                25      1           DEM  prefeito   
1             61018                55      1           PSD  prefeito   
2             61018             10999      1  REPUBLICANOS  vereador   
3             61018             10242      1  REPUBLICANOS  vereador   
4             61018             10333      1  REPUBLICANOS  vereador   

    resultado  votos  
0      eleito  11863  
1  nao eleito   3367  
2  nao eleito     28  
3  nao eleito     33  
4  nao eleito    353

In [20]:
# verificando o tamanho do dataframe

selecao_resultados_candidatos_SP.shape

(85770, 7)

In [21]:
# verificando os cargos para o estado de SP com os dados colhidos

selecao_resultados_candidatos_SP.cargo.unique()

array(['prefeito', 'vereador'], dtype=object)

In [22]:
# escolhendo somente os dados para as prefeituras do estado de SP

selecao_prefeitos_resultados_candidatos_SP = selecao_resultados_candidatos_SP.query('cargo == "prefeito"')

# vizualizando os dados coletados
selecao_prefeitos_resultados_candidatos_SP.head()

id_municipio_tse  numero_candidato  turno sigla_partido     cargo  \
0               61018                25      1           DEM  prefeito   
1               61018                55      1           PSD  prefeito   
82              61034                15      1           MDB  prefeito   
83              61034                45      1          PSDB  prefeito   
128             61050                10      1  REPUBLICANOS  prefeito   

      resultado  votos  
0        eleito  11863  
1    nao eleito   3367  
82   nao eleito   1217  
83       eleito   2087  
128  nao eleito    902

In [23]:
# visualizando a quantidade de daos do dataframe com os resultados dos prefeitos dos municipios do estado de SP
selecao_prefeitos_resultados_candidatos_SP.shape

(2484, 7)

***

## 2. <font color=red>ANALISANDO OS DADOS</font>

In [24]:
# visualizando os dataframes trabalhados até o momento

display(selecao_prefeitos_receitas_candidatos_SP.head())

display(municipios_SP_codigo_tse.head())

display(selecao_prefeitos_resultados_candidatos_SP.head())

id_municipio_tse  turno  numero_candidato  \
5              63134      1                45   
6              68578      1                13   
13             62910      1                23   
26             68098      1                45   
27             63894      1                13   

                      nome_candidato     cargo  \
5     MARCO AURELIO DOS SANTOS NEVES  prefeito   
6             GEREMIAS RIBEIRO PINTO  prefeito   
13  VANDRÉ LUIZ DE CAMARGO VON ZUBEN  prefeito   
26                CARLOS HIROCI OUTI  prefeito   
27             CÉLIA REGINA BRANDANI  prefeito   

                               nome_partido sigla_partido  \
5   Partido da Social Democracia Brasileira          PSDB   
6                 Partido dos Trabalhadores            PT   
13                                Cidadania     CIDADANIA   
26  Partido da Social Democracia Brasileira          PSDB   
27                Partido dos Trabalhadores            PT   

                                descricao_receita  valor_receita  
5                         VENTILADORES VENTISILVA         178.03  
6   FORD FOCUS SEDAN 2010/2011 EM NOME DO CONJUGE        1000.00  
13                                            NaN       20000.00  
26                                            NaN       12000.00  
27                              MESA DE PVC USADA          55.00

codigo_tse  uf nome_municipio  capital  codigo_ibge
0       61018  SP     ADAMANTINA        0      3500105
1       61034  SP         ADOLFO        0      3500204
2       61050  SP          AGUAÍ        0      3500303
3       61131  SP         AGUDOS        0      3500709
4       61581  SP       ALAMBARI        0      3500758

id_municipio_tse  numero_candidato  turno sigla_partido     cargo  \
0               61018                25      1           DEM  prefeito   
1               61018                55      1           PSD  prefeito   
82              61034                15      1           MDB  prefeito   
83              61034                45      1          PSDB  prefeito   
128             61050                10      1  REPUBLICANOS  prefeito   

      resultado  votos  
0        eleito  11863  
1    nao eleito   3367  
82   nao eleito   1217  
83       eleito   2087  
128  nao eleito    902

### 2.1 <font color=green>Construindo o dataframe com os nomes dos municípios, os resultados de cada candidato e a receita de cada candidato em função dos municípios e dos turnos</font>

***

#### 2.1.1 <font color=blue>Construindo o dataframe com os nomes dos municípios e os resultados de cada candidato em função dos municípios e dos turnos</font>

In [25]:
# juntando os dataframes com os nomes dos municípios e os dados dos resultados de cada município segundo o 
# código do tse pra cada município de SP
selecao_resultados_prefeitos_municipios_SP = pd.merge( 
    left = municipios_SP_codigo_tse[['codigo_tse', 'nome_municipio']], # selecionando o dataframe com o código dos municípios
    right = selecao_prefeitos_resultados_candidatos_SP, # selecionando o dataframe com os resultados das eleições
    left_on = 'codigo_tse', # selecionando a chave comum em que será feita a junção no dataframe da esquerda
    right_on = 'id_municipio_tse' # selecionando a chave comum em que será feita a junção no dataframe da direita
).drop('id_municipio_tse', axis=1) # excluindo a coluna com o código do município segundo o tse

# visualizando o dataframe criado
selecao_resultados_prefeitos_municipios_SP.head()

codigo_tse nome_municipio  numero_candidato  turno sigla_partido     cargo  \
0       61018     ADAMANTINA                25      1           DEM  prefeito   
1       61018     ADAMANTINA                55      1           PSD  prefeito   
2       61034         ADOLFO                15      1           MDB  prefeito   
3       61034         ADOLFO                45      1          PSDB  prefeito   
4       61050          AGUAÍ                10      1  REPUBLICANOS  prefeito   

    resultado  votos  
0      eleito  11863  
1  nao eleito   3367  
2  nao eleito   1217  
3      eleito   2087  
4  nao eleito    902

#### 2.1.2 <font color=blue>Construindo o dataframe com os nomes dos municípios e as receitas de cada candidato em função dos municípios e dos turnos</font>

In [26]:
# verificando a receita para cada candidato
selecao_prefeitos_receitas_candidatos_SP[['numero_candidato','valor_receita','turno']].sort_values('numero_candidato')  

numero_candidato  valor_receita  turno
57192                 10         1112.0      1
321512                10         3000.0      1
188952                10        10000.0      2
145345                10         1800.0      1
321552                10          691.7      1
...                  ...            ...    ...
232980                90         1500.0      1
214709                90         1000.0      1
54449                 90         1000.0      1
189102                90         5000.0      1
73123                 90         1163.0      1

[36226 rows x 3 columns]

Com o resultado anterior verificou-se que <b>o dataframe com as receitas de cada candidato apresentava diversos valores de receitas para um mesmo candidato</b>. A partir deste resultado, foi feito o agrupamento das receitas para prosseguir com a análise 

In [27]:
# agrupando a receita de cada candidato em função de cada turno e município
selecao_prefeitos_receitas_turnos_candidatos_SP = selecao_prefeitos_receitas_candidatos_SP.groupby(['id_municipio_tse','numero_candidato','turno'])[['valor_receita']].sum().reset_index()

# vizualizando o dataframe criado
selecao_prefeitos_receitas_turnos_candidatos_SP.head()

id_municipio_tse  numero_candidato  turno  valor_receita
0             61000                10      1        47000.0
1             61000                45      1        14613.0
2             61018                25      1        64371.0
3             61018                55      1        14700.0
4             61026                15      1        24400.0

In [28]:
# juntando os dataframes com os nomes dos municípios as receita agrupadas dos candidatos
selecao_receitas_prefeitos_municipios_SP = pd.merge(
    left = municipios_SP_codigo_tse[['codigo_tse','nome_municipio']], # selecionando o dataframe com o código dos municípios
    right =  selecao_prefeitos_receitas_turnos_candidatos_SP, # selecionando o dataframe com as receitas agrupadas dos candidatos
    left_on = 'codigo_tse', # selecionando a chave comum em que será feita a junção no dataframe da esquerda
    right_on = 'id_municipio_tse' # selecionando a chave comum em que será feita a junção no dataframe da direita
).drop('id_municipio_tse', axis=1) # excluindo a coluna com o código do município segundo o tse

# vizualizando o dataframe criado
selecao_receitas_prefeitos_municipios_SP.head()

codigo_tse nome_municipio  numero_candidato  turno  valor_receita
0       61018     ADAMANTINA                25      1        64371.0
1       61018     ADAMANTINA                55      1        14700.0
2       61034         ADOLFO                15      1        14800.0
3       61034         ADOLFO                45      1        29758.0
4       61050          AGUAÍ                10      1        19600.0

In [29]:
# verificando os orçamentos no segundo turno
selecao_prefeitos_receitas_turnos_candidatos_SP.query('turno == 2')

id_municipio_tse  numero_candidato  turno  valor_receita
422              62197                25      2      592000.00
596              62910                10      2     5196547.38
601              62910                22      2     7037508.99
775              63770                13      2     1996437.78
784              63770                55      2      347434.23
875              64254                55      2      376692.00
968              64777                13      2     3277576.47
974              64777                55      2     2527200.00
1307             66397                19      2      957780.00
1417             66893                40      2     1276286.48
1465             67130                45      2     1859657.60
1763             68756                25      2      190650.17
1766             68756                45      2      330194.00
1849             69213                45      2     2404500.00
1945             69698                40      2     1686540.00
1946             69698                45      2     3173800.00
2236             71072                45      2    20461288.72
2237             71072                50      2     7482333.41
2272             71218                45      2     2167208.19
2311             71455                10      2     2132106.76
2313             71455                17      2     2572500.00
2342             71579                19      2     1187000.00
2344             71579                45      2     1395000.00
2390             71838                15      2      882500.00
2393             71838                23      2      991452.03

Com o resultado anterior, verificou-se que <b>alguns municípios apresentaram orçamento pra somente um único candidato no segundo turno</b>.

#### 2.1.3 <font color=blue>Juntando os dataframes criados</font>

In [30]:
# juntando os dataframes com os nomes dos municípios, as receita e agrupadas os resultados dos candidatos
selecao_receitas_resultados_prefeitos_municipios_SP = pd.merge(
    left = selecao_resultados_prefeitos_municipios_SP, # selecionando o dataframe com os resultados dos candidatos
    right = selecao_receitas_prefeitos_municipios_SP, # selecionando o dataframe com as receitas dos candidatos
    on = ['codigo_tse', 'nome_municipio','numero_candidato', 'turno'] # selecionando as chaves comum em que serão feitas as junções dos dataframes
)

# vizualizando o dataframe criado
selecao_receitas_resultados_prefeitos_municipios_SP

codigo_tse nome_municipio  numero_candidato  turno sigla_partido  \
0          61018     ADAMANTINA                25      1           DEM   
1          61018     ADAMANTINA                55      1           PSD   
2          61034         ADOLFO                15      1           MDB   
3          61034         ADOLFO                45      1          PSDB   
4          61050          AGUAÍ                10      1  REPUBLICANOS   
...          ...            ...               ...    ...           ...   
2368       67776           ÓLEO                14      1           PTB   
2369       67776           ÓLEO                15      1           MDB   
2370       67776           ÓLEO                19      1          PODE   
2371       67776           ÓLEO                40      1           PSB   
2372       67776           ÓLEO                45      1          PSDB   

         cargo   resultado  votos  valor_receita  
0     prefeito      eleito  11863        64371.0  
1     prefeito  nao eleito   3367        14700.0  
2     prefeito  nao eleito   1217        14800.0  
3     prefeito      eleito   2087        29758.0  
4     prefeito  nao eleito    902        19600.0  
...        ...         ...    ...            ...  
2368  prefeito  nao eleito    619         4797.3  
2369  prefeito  nao eleito    332         7300.0  
2370  prefeito  nao eleito     85          600.0  
2371  prefeito  nao eleito    315         3000.0  
2372  prefeito      eleito    666         3585.5  

[2373 rows x 9 columns]

### 2.2 <font color=green>Fazendo a análise para os dados do 1° turno</font>

***

In [31]:
# selecionando os dados do 1° turno
selecao_receitas_resultados_prefeitos_municipios_SP_1o_turno = selecao_receitas_resultados_prefeitos_municipios_SP.query('turno == 1')

# vizualizando o dataframe criado
selecao_receitas_resultados_prefeitos_municipios_SP_1o_turno.head()

codigo_tse nome_municipio  numero_candidato  turno sigla_partido     cargo  \
0       61018     ADAMANTINA                25      1           DEM  prefeito   
1       61018     ADAMANTINA                55      1           PSD  prefeito   
2       61034         ADOLFO                15      1           MDB  prefeito   
3       61034         ADOLFO                45      1          PSDB  prefeito   
4       61050          AGUAÍ                10      1  REPUBLICANOS  prefeito   

    resultado  votos  valor_receita  
0      eleito  11863        64371.0  
1  nao eleito   3367        14700.0  
2  nao eleito   1217        14800.0  
3      eleito   2087        29758.0  
4  nao eleito    902        19600.0

In [32]:
# criando uma coluna com a receita por voto de cada candidato (eleito ou não)
selecao_receitas_resultados_prefeitos_municipios_SP_1o_turno = selecao_receitas_resultados_prefeitos_municipios_SP_1o_turno.assign(receita_por_voto=selecao_receitas_resultados_prefeitos_municipios_SP_1o_turno['valor_receita']/selecao_receitas_resultados_prefeitos_municipios_SP_1o_turno['votos']) 

In [33]:
selecao_receitas_resultados_prefeitos_municipios_SP_1o_turno

codigo_tse nome_municipio  numero_candidato  turno sigla_partido  \
0          61018     ADAMANTINA                25      1           DEM   
1          61018     ADAMANTINA                55      1           PSD   
2          61034         ADOLFO                15      1           MDB   
3          61034         ADOLFO                45      1          PSDB   
4          61050          AGUAÍ                10      1  REPUBLICANOS   
...          ...            ...               ...    ...           ...   
2368       67776           ÓLEO                14      1           PTB   
2369       67776           ÓLEO                15      1           MDB   
2370       67776           ÓLEO                19      1          PODE   
2371       67776           ÓLEO                40      1           PSB   
2372       67776           ÓLEO                45      1          PSDB   

         cargo   resultado  votos  valor_receita  receita_por_voto  
0     prefeito      eleito  11863        64371.0          5.426199  
1     prefeito  nao eleito   3367        14700.0          4.365904  
2     prefeito  nao eleito   1217        14800.0         12.161052  
3     prefeito      eleito   2087        29758.0         14.258745  
4     prefeito  nao eleito    902        19600.0         21.729490  
...        ...         ...    ...            ...               ...  
2368  prefeito  nao eleito    619         4797.3          7.750081  
2369  prefeito  nao eleito    332         7300.0         21.987952  
2370  prefeito  nao eleito     85          600.0          7.058824  
2371  prefeito  nao eleito    315         3000.0          9.523810  
2372  prefeito      eleito    666         3585.5          5.383634  

[2349 rows x 10 columns]

In [34]:
selecao_receitas_resultados_prefeitos_municipios_SP_1o_turno.query('resultado == "eleito"').receita_por_voto.describe()

count    610.000000
mean      16.201118
std       11.512425
min        0.145045
25%        8.870842
50%       13.736509
75%       20.287784
max      117.190866
Name: receita_por_voto, dtype: float64

Para <font color=blue><b>os candidatos eleitos</b></font> temos:

### $
\begin{cases}
média_{eleitos} = 16.20\\
\sigma_{eleitos} = 11.51\\
\sigma_{méd_{eleitos}} = 0.47
\end{cases}
$

In [35]:
selecao_receitas_resultados_prefeitos_municipios_SP_1o_turno.query('resultado == "nao eleito"').receita_por_voto.describe()

count    1733.000000
mean       32.608213
std        47.275515
min         0.132957
25%        10.457335
50%        19.323572
75%        37.592874
max       810.810811
Name: receita_por_voto, dtype: float64

Para <font color=red><b>os candidatos não eleitos</b></font> temos:

### $
\begin{cases}
média_{não\ eleitos} = 32.61\\
\sigma_{não\ eleitos} = 47.28\\
\sigma_{méd_{não\ eleitos}} = 1.34
\end{cases}
$